src/containers/Home/components/ProductForm/index.jsx

```diff
@@ -0,0 +1,178 @@
+import React, { useState } from 'react';
+import { toast } from 'react-hot-toast';
+import { useProductContext } from '../../../../contexts/ProductContext';
+import Form from '../MainForm';
+
+const ProductForm = () => {
+  const [showAddForm, setShowAddForm] = useState(false);
+  const [showEditForm, setShowEditForm] = useState(false);
+  const [fetched, setFetched] = useState(false);
+  const { state, dispatch } = useProductContext();
+  const { products } = state;
+  const [formData, setFormData] = useState({
+    title: '',
+    price: '',
+    description: '',
+    image: null,
+  });
+
+  const handleChange = (e) => {
+    const { name, value } = e.target;
+    e.target.classList.contains('is-invalid') &&
+      e.target.classList.remove('is-invalid');
+    e.target.classList.contains('is-valid') &&
+      e.target.classList.remove('is-valid');
+    if (name === 'price' && value <= 0) {
+      e.target.classList.add('is-invalid');
+      toast.error('Price must be greater than 0.');
+      return;
+    } else if (value === '') {
+      e.target.classList.add('is-invalid');
+      toast.error('Please fill in all fields.');
+      return;
+    }
+    e.target.classList.add('is-valid');
+    setFormData({ ...formData, [name]: value });
+  };
+
+  const handleImageChange = (e) => {
+    const file = e.target.files[0];
+    e.target.classList.contains('is-invalid') &&
+      e.target.classList.remove('is-invalid');
+    e.target.classList.contains('is-valid') &&
+      e.target.classList.remove('is-valid');
+    const reader = new FileReader();
+    reader.readAsDataURL(file);
+    reader.onload = (innerEvent) => {
+      const img = new Image();
+      img.src = innerEvent.target.result;
+      img.onload = () => {
+        if (img.width > 200 || img.height > 200) {
+          toast.error('Image must be 200px x 200px or smaller.');
+          e.target.classList.add('is-invalid');
+          return;
+        }
+      };
+    };
+    e.target.classList.add('is-valid');
+    setFormData({ ...formData, image: URL.createObjectURL(file) });
+  };
+
+  const handleSubmit = (e) => {
+    e.preventDefault();
+    if (
+      formData.title.trim() === '' ||
+      formData.description.trim() === '' ||
+      !formData.image
+    ) {
+      toast.error('Please fill in all fields.');
+      return;
+    }
+
+    if (formData.id !== undefined) {
+      dispatch({ type: 'UPDATE_PRODUCT', payload: formData });
+      toast.success('Product updated successfully!');
+      setFormData({
+        title: '',
+        price: '',
+        image: null,
+        description: '',
+      });
```

Comment:
Instead of repeating the same thing for add and edit, we can make a separate function to reset this data.